### PR TITLE
refactor(finance): rename mask to maskedNumber

### PR DESCRIPTION
**Pull request by Shinigami92**
references https://github.com/faker-js/faker/issues/1346#issuecomment-1407724091

### CODE REVIEW

File: src/modules/finance/index.ts [hunk 1]
```diff
@@ -274,18 +292,17 @@ export class FinanceModule {
...
+   * faker.finance.maskNumber(3) // '(...342)'
+   * faker.finance.maskNumber({ length: 3 }) // '(...342)'
+   * faker.finance.maskNumber({ length: 3, parens: false }) // '...236'
+   * faker.finance.maskNumber({ length: 3, parens: false, ellipsis: false }) // '298'
    *
-   * @since 2.0.1
+   * @since 8.0.0
    */
-  mask(
+  maskNumber(
```
Comment (ST-DDT) [inline src/modules/finance/index.ts]: Maybe name this `maskedNumber`? Because it doesn't mask a given number, but masks a new one.

**Review by ST-DDT — APPROVED** [review]

**Review by ST-DDT — CHANGES_REQUESTED** [review]
Please include this in the upgrading guide.

File: src/modules/finance/index.ts [hunk 2]
```diff
@@ -197,8 +198,31 @@ export class FinanceModule {
    * faker.finance.mask(3, false, false) // '298'
    *
    * @since 2.0.1
+   *
+   * @deprecated Use `faker.finance.maskedNumber` instead.
```
Comment (ST-DDT) [inline src/modules/finance/index.ts:230]: This should also have a `@see` tag.

**Comment by matthewmayer** [comment]
> Please include this in the upgrading guide.

i dont think the upgrading guide has to mention every deprecation, only
- breaking changes
- major module renames like name to person
- unexpected behavior changes

if a method is just changed to a better name within the same module, and the deprecation workflow is used, i think its fine to just mention this as a breaking change in the v9 upgrading guide when it is actually removed.

**Comment by ST-DDT** [comment]
> i dont think the upgrading guide has to mention every deprecation

Not individually, but maybe as a list of renamed methods.

**Review by ST-DDT — DISMISSED** [review]

**Review by xDivisionByZerox — DISMISSED** [review]

**Comment by Shinigami92** [comment]
@matthewmayer @ST-DDT I cannot find it right now, but I thought we have an open TODO/issue to re-evaluate the upgrading guideline in total before releasing v8.0
Then we should check everything over and unify some stuff and so on (in the upgrading.md)

**Comment by matthewmayer** [comment]
It would be good to have some guidelines on what needs to be in upgrading and what doesn't per my other comment.

**Comment by Shinigami92** [comment]
> It would be good to have some guidelines on what needs to be in upgrading and what doesn't per my other comment.

I suggest moving this into its own discussion, discord chat or even issue

**Comment by matthewmayer** [comment]
Let's make an issue

**Review by xDivisionByZerox — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**